Documentation/special-services/pod-mutator.md

```diff
@@ -0,0 +1,74 @@
+# PodMutator Service
+To provide pod mutation capabilities for finer-grained control over rkt containerization concepts and debugging needs, 
+this proposal introduces a new optional stage1 service with special semantics that is lightweight.
+
+When added to a pod manifest, the PodMutator service, which is just an app by itself, will expose app-level operations 
+to the outside using RPC by communicating with the systemd inside the pod.
+
+PodMutator will expose this interface through a special network namespace that will 
+only be used by special services such as itself (use default-restricted namespace?).
+
+## Architecture
+
+PodMutator, in essence, is rkt application container manipulation logic on top systemd/systemd-nspawn calls.
+Meaning that starting an App is translated into pod systemd calls to start that app's services.
+
+![pod-mutator](pod-mutator.png)
+
+## Interface
+
+```go
+type AppID string
+
+type AppState string
+
+const (
+	FailureAppState string = "Fail"
+	SuccessAppState string = "Success"
+	IdleAppState    string = "Idle"
+)
+
+type AppStatus struct {
+	ID          AppID
+	Name        string
+	State       AppState
+	Labels      []string
+	Annotations []string
+	Metadata    struct {
```

Comment:
These doesn't look like metadata, fwiw, the k8s' metadata includes "id, name, annotation"